config/routes.rb

```diff
@@ -1,6 +1,13 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  devise_for :users
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  devise_scope :user do
+    delete 'logout', to: 'devise/sessions#destroy', as: :logout
+  end
+  
+  root to: "users#index"
+  
+  resources :groups do
+    resources :entities
+  end
```

Comment:
- [Optional] Based on the requirements, we were supposed to only add necessary routes corresponding  to the controller actions we implemented. By using resourceful routing, 7 routes are generated by default while you didn't implement all the actions.